frontend/src/bundles/common/components/input/input.tsx

```diff
@@ -1,41 +1,26 @@
-import {
-    type Control,
-    type FieldErrors,
-    type FieldPath,
-    type FieldValues,
-} from 'react-hook-form';
+import clsx from 'clsx';
+import { type InputHTMLAttributes, forwardRef } from 'react';
 
-import { useFormController } from '~/bundles/common/hooks/hooks.js';
+import styles from './styles.module.scss';
 
-type Properties<T extends FieldValues> = {
-    control: Control<T, null>;
-    errors: FieldErrors<T>;
-    label: string;
-    name: FieldPath<T>;
-    placeholder?: string;
-    type?: 'text' | 'email';
-};
-
-const Input = <T extends FieldValues>({
-    control,
-    errors,
-    label,
-    name,
-    placeholder = '',
-    type = 'text',
-}: Properties<T>): JSX.Element => {
-    const { field } = useFormController({ name, control });
-
-    const error = errors[name]?.message;
-    const hasError = Boolean(error);
-
-    return (
-        <label>
-            <span>{label}</span>
-            <input {...field} type={type} placeholder={placeholder} />
-            {hasError && <span>{error as string}</span>}
-        </label>
-    );
-};
+interface Properties extends InputHTMLAttributes<HTMLInputElement> {
+    hasError?: boolean;
+    width?: string;
+}
+const Input = forwardRef<HTMLInputElement, Properties>(
+    ({ hasError = false, width = 'auto', disabled, ...otherProperties }, reference) => (
+        <input
+            className={clsx(styles.input, {
+                [styles.input__error]: hasError,
+                [styles.input__disabled]: disabled,
+            })}
+            style={{ width }}
+            ref={reference}
+            type="text"
+            {...otherProperties}
+        />
+    ),
+);
 
+Input.displayName = 'Input';
```

Comment:
Why do we need this?

Reply:
https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/display-name.md